### PR TITLE
Add --frozen-lockfile to fail if Lockfile needs update

### PR DESCRIPTION
**Pull request by lawliet89**
**Summary**

Adds the flag `--frozen-lockfile` to `yarn install` so that `yarn` will fail if it detects that the lockfile requires an update. This is similar to [bundler's `--frozen` flag](https://bundler.io/v1.13/man/bundle-install.1.html).

**Test plan**

Consider the `package.json` and `yarn.lock` in the fixture I have added in this PR. 

```
$ yarn --frozen-lockfile
yarn install v0.18.0-0
info No lockfile found.
[1/4] Resolving packages...
error Your lockfile needs to be updated, but yarn was run with `--frozen-lockfile`
Done in 6.28s.

```

But if you generate a lockfile and re-run `yarn`, it should install fine.

### CODE REVIEW

File: src/reporters/lang/en.js [hunk 1]
```diff
@@ -88,6 +88,7 @@ const messages = {
   jsonError: 'Error parsing JSON at $0, $1.',
   noFilePermission: "We don't have permissions to touch the file $0.",
   allDependenciesUpToDate: 'All of your dependencies are up to date.',
+  frozenLockfileError: 'Your lockfile needs to be updated, but yarn was run with `--frozen-lockfile`',
```
Comment (kittens) [inline src/reporters/lang/en.js]: run -> ran

Comment (kittens) [inline src/reporters/lang/en.js]: Add a dot to the end of this as well since it's reporting an action.

Comment (lawliet89) [inline src/reporters/lang/en.js]: [The past participle of run is run.](https://www.quora.com/Which-is-more-grammatically-correct-was-run-or-was-ran) I will add the period.

**Comment by mmun** [comment]
@lawliet89 Thanks for implementing this feature! Looking forward to using it on our CI to catch when people forget commit their yarn.lock, or if they added a dependency with `npm install` instead of `yarn add`.

**Comment by bestander** [comment]
@lawliet89, could you rebase please?

**Comment by lawliet89** [comment]
@bestander I have rebased.

**Pull request merged**